src/wattsi.pas

```diff
@@ -1999,11 +2007,25 @@ procedure Inform(Message: UTF8String);
       Writeln(Message);
 end;
 
+// http://wiki.freepascal.org/UTF8_strings_and_characters#Search_and_copy
+function SplitInHalf(Txt, Separator: UTF8String; out Half1, Half2: UTF8String): Boolean;
+var
+  i: Integer;
+begin
+  i := Pos(Separator, Txt);
```

Comment:
Specifically, I'd use the https://github.com/whatwg/wattsi/blob/master/src/lib/stringutils.pas stuff to iterate over the string, grab the start pointer and the middle pointer using a single walk, and return the two CutUTF8Strings.

But whatever. This code would work fine too.